src/sentry/api/endpoints/external_user.py

```diff
@@ -0,0 +1,82 @@
+import logging
+
+from django.db import IntegrityError
+
+from rest_framework import serializers, status
+from rest_framework.response import Response
+from sentry.api.serializers.rest_framework.base import CamelSnakeModelSerializer
+
+from sentry.api.bases.organization import OrganizationEndpoint
+
+from sentry.api.serializers import serialize
+from sentry.models import ExternalUser, EXTERNAL_PROVIDERS, OrganizationMember
+
+logger = logging.getLogger(__name__)
+
+
+class ExternalUserSerializer(CamelSnakeModelSerializer):
+    member_id = serializers.IntegerField(required=True)
+    external_name = serializers.CharField(required=True)
+    provider = serializers.ChoiceField(choices=list(EXTERNAL_PROVIDERS.values()))
+
+    class Meta:
+        model = ExternalUser
+        fields = ["member_id", "external_name", "provider"]
+
+    def validate_provider(self, provider):
+        if provider not in EXTERNAL_PROVIDERS.values():
+            raise serializers.ValidationError(
+                f'The provider "{provider}" is not supported. We currently accept Github and Gitlab user identities.'
+            )
+        return ExternalUser.get_provider_enum(provider)
+
+    def validate_member_id(self, member_id):
+        try:
+            return OrganizationMember.objects.get(
```

Comment:
@NisanthanNanthakumar Nit: I think you can do `OrganizationMember.objects.filter(...).exists()` here to avoid actually retrieving the DB object which we aren't using.

Reply:
@scefali we use this value here: https://github.com/getsentry/sentry/blob/5209c7a28bd72e2dcd982084c4c49d000cbafd61/src/sentry/api/endpoints/external_user.py#L42